internal/jobcompleter/job_completer_test.go

```diff
@@ -25,6 +26,7 @@ import (
 )
 
 type partialExecutorMock struct {
+	riverdriver.Executor
```

Comment:
This is now a full executor, but one that allows specific methods to be overridden. A tx variant of it was also needed because of the txn change in the completer.